importer/handlers/shapefile/handler.py

```diff
@@ -151,15 +152,28 @@ def create_ogr2ogr_command(files, original_name, ovverwrite_layer, alternate):
         )
         layers = ogr.Open(files.get("base_file"))
         layer = layers.GetLayer(original_name)
-        additional_option = (
+
+        if not files.get("cpg_file") and files.get("cst_file"):
+            # GeoServer exports cst-file
+            encoding_file = files.get("cst_file")
+            with open(encoding_file) as f:
+                encoding = f.read()
+            try:
+                codecs.lookup(encoding)
+            except LookupError as e:
+                encoding = None
+                logger.error(f"Will ignore invalid encoding: {e}")
+
+        additional_options = (
             " -nlt PROMOTE_TO_MULTI"
             if layer is not None
             and "Point" not in ogr.GeometryTypeToName(layer.GetGeomType())
-            else " "
+            else " ",
+            " ENCODING={encoding} " if encoding else "",
```

Comment:
I guess this will raise an error if the above files are not shipped along with the SHP one.
In case both `cpg_file` and `cst_file` are None, the encoding variable is not generated and the if will fail.
Should be enough to set `encoding=None` at line 155